docs/source/usage/cheatsheet.rst

```diff
@@ -0,0 +1,283 @@
+       device,
+       globalThreadExtent, elementsPerThread,
+       false,
+       workdiv::GridBlockExtentSubDivRestrictions::Unrestricted);
+
+Manually set a kernel launch configuration
+  .. code-block:: c++
+
+     vec::Vec<Dim, Idx> const blocksPerGrid = vectorValue;
+     vec::Vec<Dim, Idx> const threadsPerBlock = vectorValue;
+     vec::Vec<Dim, Idx> const elementsPerThread = vectorValue;
+
+     using WorkDiv = workdiv::WorkDivMembers<Dim, Idx>;
+     auto manualWorkDiv = WorkDiv{blocksPerGrid,
+                                  threadsPerBlock,
+				  elementsPerThread};
+
+Instantiate a kernel and create a task that will run it (does not launch it yet)
+  .. code-block:: c++
+
+     Kernel kernel{argumentsForConstructor};
+     auto taskRunKernel = kernel::createTaskKernel<Acc>(workDiv,
+                                                        kernel,
+							parameters);
+
+acc parameter of the kernel is provided automatically, does not need to be specified here
+
+Put the kernel for execution
+  .. code-block:: c++
+
+     queue::enqueue(queue, taskRunKernel);
+
+Kernel Implementation
+---------------------
+
+Define a kernel as a C++ functor
+  .. code-block:: c++
+
+     struct Kernel {
+        template<typename Acc>
+        ALPAKA_FN_ACC void operator()(Acc const & acc, parameters) const { ... }
+     };
+
+``ALPAKA_FN_ACC`` is required for kernels and functions called inside, acc is mandatory first parameter, its type is the template parameter
+
+Access multi-dimensional indices and extents of blocks, threads, and elements
+  .. code-block:: c++
+
+     auto idx = idx::getIdx<Origin, Unit>(acc);
+     auto extent = workdiv::getWorkdiv<Origin, Unit>(acc);
+
+  Origin:
+     .. code-block:: c++
+
+	Grid, Block, Thread
+
+  Unit:
+     .. code-block:: c++
+
+	Blocks, Threads, Elems
+
+Access components of multi-dimensional indices and extents
+  .. code-block:: c++
+
+     auto idxX = idx[0];
+
+Linearize multi-dimensional vectors
+  .. code-block:: c++
+
+     auto linearIdx = idx::mapIdx<1u>(idx, extent);
+
+.. raw:: pdf
+
+   Spacer 0,8
+
+Allocate static shared memory variable
+  .. code-block:: c++
+
+     Type & var = block::shared::st::allocVar<Type, __COUNTER__>(acc);
+
+Get dynamic shared memory pool, requires the kernel to specialize
+  .. code-block:: c++
+
+     kernel::traits::BlockSharedMemDynSizeBytes
+       Type * dynamicSharedMemoryPool = block::shared::dyn::getMem<Type>(acc);
+
+Synchronize threads of the same block
+  .. code-block:: c++
+
+     block::sync::syncBlockThreads(acc);
+
+Atomic operations
+  .. code-block:: c++
+
+     auto result = atomic::atomicOp<Operation>(acc,
+                                               arguments,
+					       OperationHierarchy);
+
+  Operation (all in atomic::op):
+     .. code-block:: c++
+
+	namespace atomic::op {
+           Add, Sub, Min, Max, Exch, Inc, Dec, And, Or, Xor, Cas
+	}
+
+  OperationHierarchy (all in hierarchy):
```

Comment:
I like this change to put stuff in the namespace, but then the brackets (all in ...) is not necessary.